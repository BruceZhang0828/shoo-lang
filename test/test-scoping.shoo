//scoping for globals vs locals

string msg = "comment ca va";

function hello() void {
	string msg = "ca va bien";
	println(msg);
}

hello();

// globals vs parameters
function hello2(string msg) void {
	println(msg);
}

hello2("ca va mal");