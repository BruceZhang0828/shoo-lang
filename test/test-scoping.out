ca va bien
ca va mal